test/ctia/test_helpers/search.clj

```diff
@@ -152,95 +159,140 @@
             ;; OR could match other test documents matching "http"
             "escaping reserved characters should avoid parsing errors and preserve behavior of OR"))
       (is (= matched-ids
-             (search-ids search-uri (str "description:" domain-word)))
+             (search-ids entity (str "description:" domain-word)))
           "tokenization of :description field should split urls on characters like \\/ \\: and thus enables search on url components")
       (is (= matched-ids
-             (search-ids search-uri (str "description:" base-domain))
-             (search-ids search-uri base-domain))
+             (search-ids entity (str "description:" base-domain))
+             (search-ids entity base-domain))
           "word delimiter filtering of :description should split tokens on \\. and thus enables search on domain components"))
 
     ;; test number fitering
     (is (= matched-ids
-           (search-ids search-uri "description:127.0.0.1")
-           (search-ids search-uri "127.0.0.1"))
+           (search-ids entity "description:127.0.0.1")
+           (search-ids entity "127.0.0.1"))
         "word delimiter filtering should preserve ips")
-    (is (empty? (search-ids search-uri "127"))
+    (is (empty? (search-ids entity "127"))
         "word delimiter filtering should preserve ips")
     (testing "word delimiter filtering should not split words on numbers"
       (is (= matched-ids
-             (search-ids search-uri "description:j2ee")
-             (search-ids search-uri "j2ee")))
-      (is (empty? (search-ids search-uri "j OR ee"))))
+             (search-ids entity "description:j2ee")
+             (search-ids entity "j2ee")))
+      (is (empty? (search-ids entity "j OR ee"))))
 
     ;; clean
-    (doseq [_id (concat matched-ids unmatched-ids)]
-      (let [delete-uri (->> (long-id->id _id)
-                            :short-id
-                            (format "ctia/%s/%s" entity))]
-        (delete delete-uri
-                :headers {"Authorization" "45c1f5e3f05d0"})))))
+    (doseq [full-id (concat matched-ids unmatched-ids)]
+      (delete-doc entity full-id))))
+
+(defn ensure-one-document
+  [f example entity & args]
+  (let [{{full-id :id} :parsed-body} (create-doc entity (dissoc example :id))]
+       (apply (partial f entity) args)
+       (delete-doc entity full-id)))
 
 (defn test-non-describable-search
   [entity query query-field]
-  (let [search-uri (format "ctia/%s/search" (name entity))]
-    (testing (format "GET %s" search-uri)
-      (let [response (search search-uri query)]
-        (is (= 200 (:status response)))
-        (is (= query (first (map query-field (:parsed-body response))))
-            "query term works"))
+  (testing "search term filter"
+    (let [response (search-text entity query)]
+      (is (= 200 (:status response)))
+      (is (= query (first (map query-field (:parsed-body response))))
```

Comment:
Stronger assertion: `= [query] (map ...)`

Reply:
I totally agree, but I am not sure that there is only one document here, it requires a deeper refacto after iterative test additions on this ns and crud-test ns. 
I discovered here that it was not properly tested (not the purpose of that PR) and thus added this test.
I will test all the results instead with a `doseq` as suggested in your other comment.
thx!